pkg/audit/manager.go

```diff
@@ -519,16 +524,43 @@ func (am *Manager) reviewObjects(ctx context.Context, kind string, folderCount i
 			augmentedObj := target.AugmentedUnstructured{
 				Object:    *objFile,
 				Namespace: ns,
+				Source:    mutationtypes.SourceTypeOriginal,
 			}
 			resp, err := am.opa.Review(ctx, augmentedObj)
 			if err != nil {
 				am.log.Error(err, "Unable to review object from file", "fileName", fileName, "objNs", objNs)
 				continue
 			}
 
-			results := ToResults(&augmentedObj.Object, resp)
+			// Expand object and review any resultant resources
+			base := &mutationtypes.Mutable{
+				Object:    objFile,
+				Namespace: ns,
+				Username:  "",
+				Source:    mutationtypes.SourceTypeOriginal,
+			}
+			resultants, err := am.expansionSystem.Expand(base)
```

Comment:
For audit, does it make sense to expand for parent resources that already have generated resources created by K8s? If the expanded resources already exist in the cluster, then audit as is would catch those violations. The only special case is when replica is 0 right?

Reply:
We don't throw a violation for the expanded resource, the violation is associated with the generator resource (e.g. the violation is against the Deployment, not the fake Pod).

IMO this is helpful since it can track down the root cause of a violation.

Reply:
yes super helpful for the 0 replica scenario. but for replica > 0, wouldn't the current audit catch the pods that k8s created already? hence I was saying expansion is not needed for replica > 0.

Reply:
It makes it easier to track down what's creating the pods as well (e.g. is it a stateful set? Deployment? what's the resource called?)

I agree it would yield two violations, but there are also two objects involved in the violation.

Reply:
> it would yield two violations

yea, lets think about this experience a bit more. I can see how that could confuse people now they are getting way more violations.

Reply:
> yea, lets think about this experience a bit more. I can see how that could confuse people now they are getting way more violations.

Is this a blocker for merging this PR? Since this is still in alpha, we are still able to change this if we want.

FWIW there may be more violations, but that seems appropriate as every violation is associated with a resource that contributes to the violation.

Reply:
not a blocker